### ISSUE TITLE
String interpolated style in separate template produces false error

### ISSUE THREAD

**Issue by pstricks-fans** [angular/angular#49476]
### Which @angular/* package(s) are the source of the bug?

Don't known / other

### Is this a regression?

Yes

### Description

A string interpolated style below
`<p style="font-size: {{ true ? '50px' : '100px' }}">Hello</p>`

- produces false error in a separate template (`xxx.component.html`) 
- does NOT produce false error in inline template.

Both render correctly on the browser.

# Separate Template (produce false error)
```ts
import { Component } from '@angular/core';
@Component({
  selector: 'app-root',
  templateUrl: 'app.component.html'
})
export class AppComponent {}
```
```html
<p style="font-size: {{ true ? '50px' : '100px' }}">Hello</p>
```
![image](https://user-images.githubusercontent.com/50270635/226153691-22b6cef9-efbf-47c6-b70a-a1520480c088.png)

# Inline Template (no problem)
```ts
import { Component } from '@angular/core';

@Component({
  selector: 'app-root',
  template: ` <p style="font-size: {{ true ? '50px' : '100px' }}">Hello</p> `,
})
export class AppComponent {}
```

**Comment by JeanMeche** [comment]
That error is triggered by VSCode. 

You can disable it by having : `"html.validate.styles": false` in your settings (`.vscode/setting.json` or `~/.vscode/settings.json`)

**Comment by iam-ashish** [comment]
You can use property binding instead which will be easy to bind the typescript property to HTML.
[ngClass] = "true ? 'font-size: 50px' : 'font-size: 100px' "

**Comment by pstricks-fans** [comment]
@iam-ashish : Yes. I already knew that we can use one of the following permutations.

```html
<p style="{{ true ? 'font-size: 50px' : 'font-size: 100px' }}"></p>
<p style="font-size: {{ true ? '50px' : '100px' }}"></p>

<p [style]="true ? 'font-size: 50px' : 'font-size: 100px'"></p>
<p [style]="'font-size:' + (true ? '50px' : '100px')"></p>
<p [style.font-size]="true ? '50px' : '100px'"></p>
<p [style.font-size.px]="true ? 50 : 100"></p>
<p [style]="{ 'font-size': true ? '50px' : '100px' }"></p>

<p [ngStyle]="{ 'font-size': true ? '50px' : '100px' }"></p>
<p [ngStyle]="{ 'font-size.px': true ? 50 : 100 }"></p>
```

**Comment by atscott** [comment]
As mentioned by @JeanMeche, this is caused by another extension in vscode that’s out of our control https://github.com/angular/angular/issues/49476#issuecomment-1475223853.

**Comment by luca9256** [comment]
Ok meta dev luc llc ®™©Temat: Re: [angular/angular] String interpolated style in separate template produces false error (Issue #49476)Data: 2023-03-20 4:13Nadawca: "Andrew Scott" ***@***.***&gt;Adresat: "angular/angular" ***@***.***&gt;; DW: "Subscribed" ***@***.***&gt;; 

Closed #49476 as not planned.
